core/src/com/unciv/logic/multiplayer/storage/MultiplayerServer.kt

```diff
@@ -24,6 +26,14 @@ class MultiplayerServer(
     private var authenticationHeader: Map<String, String>? = null
 ) {
     internal var featureSet = ServerFeatureSet()
+        set(value) {
+            if (field != value) {
+                EventBus.send(ServerFeatureSetChanged(field, value))
```

Comment:
Please avoid setters, in favor of explicit calls when setting the value (in this case in MultiplayerServer.kt)
This allows for better ordering (first set then update / first update then set) and better error handling

Getters and setters have their place, but I also dislike them for another reason - they **hide information** in that in reading the code, it looks like nothing else is happening so there should be no reason it could fail.

Reply:
Explained above. Waiting for further feedback.

Reply:
Another way to do it would be to remove the events and place the reconnect calls inside setters.